typescript/sdk/src/token/RadixWarpRouteReader.ts

```diff
@@ -0,0 +1,162 @@
+import { RadixSDK, RadixTokenTypes } from '@hyperlane-xyz/radix-sdk';
+import { Address, assert, rootLogger } from '@hyperlane-xyz/utils';
+
+import { RadixIsmReader } from '../ism/RadixIsmReader.js';
+import { ChainMetadataManager } from '../metadata/ChainMetadataManager.js';
+import {
+  DestinationGas,
+  MailboxClientConfig,
+  RemoteRouters,
+  RemoteRoutersSchema,
+} from '../router/types.js';
+import { ChainNameOrId } from '../types.js';
+
+import { TokenType } from './config.js';
+import { DerivedTokenRouterConfig, HypTokenConfig } from './types.js';
+
+export class RadixWarpRouteReader {
+  protected readonly logger = rootLogger.child({
+    module: 'RadixWarpRouteReader',
+  });
+  ismReader: RadixIsmReader;
+
+  constructor(
+    protected readonly metadataManager: ChainMetadataManager,
+    protected readonly chain: ChainNameOrId,
+    protected readonly signer: RadixSDK,
+  ) {
+    this.ismReader = new RadixIsmReader(metadataManager, signer);
+  }
+
+  /**
+   * Derives the configuration for a Hyperlane ERC20 router contract at the given address.
+   *
+   * @param warpRouteAddress - The address of the Hyperlane ERC20 router contract.
+   * @returns The configuration for the Hyperlane ERC20 router.
+   *
+   */
+  async deriveWarpRouteConfig(
+    warpRouteAddress: Address,
+  ): Promise<DerivedTokenRouterConfig> {
+    // Derive the config type
+    const type = await this.deriveTokenType(warpRouteAddress);
+    const baseMetadata = await this.fetchMailboxClientConfig(warpRouteAddress);
+    const tokenConfig = await this.fetchTokenConfig(type, warpRouteAddress);
+    const remoteRouters = await this.fetchRemoteRouters(warpRouteAddress);
+    const destinationGas = await this.fetchDestinationGas(warpRouteAddress);
+
+    return {
+      ...baseMetadata,
+      ...tokenConfig,
+      remoteRouters,
+      destinationGas,
+      type,
+    } as DerivedTokenRouterConfig;
```

Comment:
Considering that this is new code can we try to avoid this type assertion as it makes it more difficult to find type errors? I'd use the schema to parse the object so we can be sure the data type is in the expected shape instead